src/java/org/apache/cassandra/db/guardrails/EnableFlag.java

```diff
@@ -24,35 +24,35 @@
 import org.apache.cassandra.service.ClientState;
 
 /**
- * A guardrail that completely disables the use of a particular feature.
+ * A guardrail that enables the use of a particular feature.
  *
- * <p>Note that this guardrail only aborts operations (if the feature is disabled) so is only meant for
+ * <p>Note that this guardrail only enables operations (if the feature is enables) so is only meant for
  * query-based guardrails (we're happy to reject queries deemed dangerous, but we don't want to create a guardrail
  * that breaks compaction for instance).
  */
-public class DisableFlag extends Guardrail
+public class EnableFlag extends Guardrail
 {
-    private final Predicate<ClientState> disabled;
+    private final Predicate<ClientState> enabled;
     private final String what;
 
     /**
-     * Creates a new {@link DisableFlag} guardrail.
+     * Creates a new {@link EnableFlag} guardrail.
      *
      * @param name     the identifying name of the guardrail
-     * @param disabled a {@link ClientState}-based supplier of boolean indicating whether the feature guarded by this
-     *                 guardrail must be disabled.
+     * @param enabled a {@link ClientState}-based supplier of boolean indicating whether the feature guarded by this
+     *                 guardrail is enabled.
      * @param what     The feature that is guarded by this guardrail (for reporting in error messages),
```

Comment:
I know it is not part of the patch... it just feels a bit unclear to use `what` as a var name. Probably use `featureName`.